src/Agent.Worker/TaskConditionsEvaluator.cs

```diff
@@ -0,0 +1,50 @@
+using Microsoft.TeamFoundation.DistributedTask.WebApi;
+using Microsoft.VisualStudio.Services.Agent.Util;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker
+{
+    public static class TaskConditionsEvaluator
+    {
+        public static bool AreConditionsSatisfied(IList<TaskCondition> conditions, Variables variables, IExecutionContext executionContext)
```

Comment:
Yeah this is only being used by StepsRunner. Having said that, this was being treated more like a module on its own and the L0 tests were able to test it extensively in isolation. Making it a private method within StepsRunner would increase the cost of setting up the tests, wouldn't it.. Thoughts ?